### ISSUE TITLE
Remote-write should check that label names are valid

### ISSUE THREAD

**Issue by bboreham** [prometheus/prometheus#11571]
### Proposal

Prometheus remote-write receiver should check that labels conform to [the documented restrictions](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels).
Not checking allows callers to create series which cannot be queried via PromQL.

To illustrate, change the test data for `TestRemoteWriteHandler` and the test still passes:
```
@@ -30,7 +30,7 @@ var writeRequestFixture = &prompb.WriteRequest{
        Timeseries: []prompb.TimeSeries{
                {
                        Labels: []prompb.Label{
-                               {Name: "__name__", Value: "test_metric1"},
+                               {Name: "__name__", Value: "test.metric1"},
                                {Name: "b", Value: "c"},
```

You can also pass the test with blank labels, no `__name__`, and pretty much everything else I tried.

Related: #11505

**Comment by roidelapluie** [comment]
Yes, this is something known. The only remote write client we support is Prometheus. I guess we have to be more careful now that 3rd parties also use the protocol.

**Comment by alexgreenbank** [comment]
Hello from the bug scrub.

Apologies for the delay in looking at this.

@alexgreenbank to look at a proposal to enforce the existing spec alongside other related issues.

**Comment by klsd-2** [comment]
Hi, I want to contribute to the issue. Can you assign me

**Comment by klsd-2** [comment]
Hi, I'd like to contribute to this issue if it's still open
